src/backends/gpu/shared_state.hpp

```diff
@@ -121,7 +125,7 @@ struct shared_state {
     void reset(fvm_value_type initial_voltage, fvm_value_type temperature_K);
 };
 
-// For debugging only:
+// For debugging only.
 std::ostream& operator<<(std::ostream& o, const shared_state& s);
```

Comment:
With the `friend` declaration above, this declaration for `operator<<` is redundant and can be removed.

Reply:
Now I look at it, the `operator<<` doesn't need to be a friend because `shared_state` is a `struct`. So instead I have removed the `friend` declaration from inside the struct.